lib/cocoapods/executable.rb

```diff
@@ -93,6 +94,21 @@ def self.popen3(bin, command, stdout, stderr)
       end
     end
 
+    def self.reader(input, output)
+      Thread.new do
+        buf = ''
+        begin
+          loop do
+            buf << input.readpartial(4096)
+            string, separator, buf = buf.rpartition(/[\r\n]/)
```

Comment:
But seriously, you are coupling the reader of the stream and the indenter unnecessarily. Suddenly the reader of the stream needs to `<<` in nice chunks where `\n` terminates the end of a chunk? I suggest you let the reader push output to the indenter in as big chunks as possible, and let the indenter do its thing.